pkg/cmd/get.go

```diff
@@ -142,9 +235,144 @@ func (o *GetOptions) Complete(args []string) error {
+	// Create a map of existing resources by UID for deduplication
+	var k8sTrimmed, kubeArchiveTrimmed bool
+	existingUIDs := make(map[string]int)
+	for idx, res := range existing {
+		existingUIDs[string(res.Resource.GetUID())] = idx
+	}
+
+	result := make([]*ResourceWithAvailability, 0, len(existing)+len(kubearchiveResources))
+	result = append(result, existing...)
+
+	for processedIdx, resource := range kubearchiveResources {
+		uid := string(resource.GetUID())
+
+		var insertIdx int
+		if k8sIdx, exists := existingUIDs[uid]; exists {
+			// Resource exists in both - mark as archived too
+			existing[k8sIdx].Archived = true
+		} else {
+			// Resource only in archive
+			resourceWithAvailability := &ResourceWithAvailability{
+				Resource:  resource,
+				InCluster: false,
+				Archived:  true,
+			}
+
+			// Insert in sorted order using binary search
+			insertIdx = o.findInsertionIndex(result, resource)
+
+			// Insert at the correct position
+			if insertIdx < len(result) {
+				result = append(result, nil) // Add space
+				copy(result[insertIdx+1:], result[insertIdx:])
+				result[insertIdx] = resourceWithAvailability
+			} else {
+				result = append(result, resourceWithAvailability)
+			}
+		}
+
+		// Trim to limit to save memory and stop processing if we have enough
+		if len(result) > o.Limit {
+			result = result[:o.Limit]
+			if insertIdx < len(result) {
+				k8sTrimmed = true
+			} else {
+				kubeArchiveTrimmed = true
+			}
+		}
+		if processedIdx > o.Limit || insertIdx > o.Limit {
+			break
+		}
+	}
+
+	return result, k8sTrimmed, kubeArchiveTrimmed
+}
+
+// findInsertionIndex finds the correct index to insert a resource to maintain sorted order (newest first)
+func (o *GetOptions) findInsertionIndex(resources []*ResourceWithAvailability, newResource *unstructured.Unstructured) int {
+	// Use binary search to find insertion point
+	idx, _ := slices.BinarySearchFunc(resources, newResource, func(existing *ResourceWithAvailability, target *unstructured.Unstructured) int {
+		existingTime := existing.Resource.GetCreationTimestamp().Time
+		targetTime := target.GetCreationTimestamp().Time
+
+		if targetTime.After(existingTime) {
+			return 1
+		} else if targetTime.Before(existingTime) {
+			return -1
+		}
+
+		// If timestamps are equal, sort by name for stable ordering
+		return strings.Compare(target.GetName(), existing.Resource.GetName())
+	})
+
+	return idx
+}
+
+// passesTimestampFilter checks if a resource passes the timestamp filtering criteria
+func (o *GetOptions) passesTimestampFilter(resource *unstructured.Unstructured) bool {
+	if o.After.IsZero() && o.Before.IsZero() {
+		return true
+	}
+
+	creationTime := resource.GetCreationTimestamp().Time
+	if !o.After.IsZero() && creationTime.Before(o.After) {
+		return false
+	}
+
+	if !o.Before.IsZero() && creationTime.After(o.Before) {
+		return false
+	}
+
+	return true
```

Comment:
I think this can all be replaced by these two lines:
```
creationTime := resource.GetCreationTimestamp().Time
return creationTime.After(o.After) && creationTime.Before(o.before)
```
I believe this logic is correct (even with zero values), and I think the existing code is wrong in that it will return true for values that equal `o.Before` or `o.After`.